h5py/h5z.pyx

```diff
@@ -96,6 +98,16 @@ def get_filter_info(int filter_code):
     return flags
 
 
+@with_phil
+def register_filter(Py_ssize_t cls_pointer_address):
+    '''(INT cls_pointer_address) => BOOL
```

Comment:
I couldn't find a better way than passing the memory address as a `int`.

It works both for passing a ctypes struct from Python and with a `ctypes.c_void_p` retrieved from a C function call (use case for `hdf5plugin` package).

Reply:
Hmm, that's kind of ugly, but I also don't have a better idea. Do you want to add those two ways to get a pointer-as-python-int to the docstring?

Reply:
`ctypes` actually converts `c_void_p` to `int` when used as function return type:

ctypes type | C type | Python type
-- | -- | --
c_void_p | void* | int or None

From [ctypes's fundamental data types](https://docs.python.org/3/library/ctypes.html#fundamental-data-types).

I've added a code snippet showing how to register a filter from a hdf5 filter plugin dynamic library.

I would not document the 2nd way about how to register a pure Python filter as done in the test, even if it's possible, it's not really something I would promote.